server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -1679,4 +1681,16 @@ static void validateTranslogDurabilitySettings(Settings requestSettings, Cluster
         }
 
     }
+
+    public static void validateIndexStoreLocality(Settings indexSettings) {
+        if (indexSettings.get(IndexModule.INDEX_STORE_LOCALITY_SETTING.getKey(), IndexModule.DataLocalityType.FULL.toString())
+            .equalsIgnoreCase(IndexModule.DataLocalityType.PARTIAL.toString())
+            && !FeatureFlags.isEnabled(FeatureFlags.WRITEABLE_REMOTE_INDEX_SETTING)) {
```

Comment:
please add the refactoring of this feature flag to be generic for tiering and writable warm (instead of being specific to writable warm) to an issue so that we dont miss this out.